packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -660,22 +660,26 @@ export class UserPool extends UserPoolBase {
    * Import an existing user pool based on its id.
    */
   public static fromUserPoolId(scope: Construct, id: string, userPoolId: string): IUserPool {
-    class Import extends UserPoolBase {
-      public readonly userPoolId = userPoolId;
-      public readonly userPoolArn = Stack.of(this).formatArn({
-        service: 'cognito-idp',
-        resource: 'userpool',
-        resourceName: userPoolId,
-      });
-    }
-    return new Import(scope, id);
+    let userPoolArn = Stack.of(scope).formatArn({
+      service: 'cognito-idp',
+      resource: 'userpool',
+      resourceName: userPoolId,
+    });
+
+    return new ImportedUserPool(scope, id, {
+      userPoolId: userPoolId,
+      userPoolArn: userPoolArn,
+    });
   }
 
   /**
    * Import an existing user pool based on its ARN.
    */
   public static fromUserPoolArn(scope: Construct, id: string, userPoolArn: string): IUserPool {
-    return UserPool.fromUserPoolId(scope, id, Stack.of(scope).parseArn(userPoolArn).resourceName!);
+    return new ImportedUserPool(scope, id, {
+      userPoolId: Stack.of(scope).parseArn(userPoolArn).resourceName!,
```

Comment:
Avoid the `!`. 

Instead add a conditional that throws an error is `resourceName` is undefined.

Reply:
Updated the code to check for resourceName + added a test.